go_expvar/tests/test_e2e.py

```diff
@@ -0,0 +1,30 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import logging
+
+import pytest
+
+from . import common
+
+log = logging.getLogger(__file__)
+
+CHECK_RATES_CUSTOM = ['go_expvar.num_calls']
+
+
+@pytest.mark.e2e
+def test_check_e2e(dd_agent_check):
+    aggregator = dd_agent_check(common.INSTANCE, times=3)
+
+    shared_tags = ['my_tag', 'expvar_url:{0}{1}'.format(common.INSTANCE['expvar_url'], common.GO_EXPVAR_URL_PATH)]
+
+    for gauge in common.CHECK_GAUGES:
+        aggregator.assert_metric(gauge, count=3, tags=shared_tags)
+    for gauge in common.MEMSTAT_PAUSE_HISTOGRAM:
+        # metric submitted only when the GC has run
```

Comment:
Can you give more details? Is that why we run the check three times and then why does running it three times trigger the GC?

Reply:
It's caused by this https://github.com/DataDog/integrations-core/blob/33db83989dd8a07c9e85ef21a6701165da28df27/go_expvar/datadog_checks/go_expvar/go_expvar.py#L110-L112

> Is that why we run the check three times and then why does running it three times trigger the GC?
Actually, it's triggered once the first time, then not triggered for the following again.